DOCS-6851-added new content (#18)
mulesoft/docs-service-mesh: modules/ROOT/pages/create-an-api-configure-service-mesh.adoc

endif::[]
:keywords: service mesh, microservices, downloading, installing

After you provision the adapter, you must create an API in Anypoint Platform to bind to the service. 

> == Methods for Creating an API

Reviewer: Priya-  my understanding based on our last conversation you will have two separate navigation flows 

1. Create an API using Anypoint Platforrm
This will cover what you have mentioned here plus  include this section - " Bind the APIs created using Anypoint Platform"

2. Create an API using service mesh auto.-creation 
This will cover what you have mentioned here plus  include this section -"Bind the APIs created using Anypoint Service Mesh CLI"

Currently, you have binding as a completely different section and missing context. There are two types of binding flows and you get to choose one based on how API is created. so it's really important these two sections are section explained together.

Author: Fixed!

Based on your requirements, you can create your API using either of the following methods:

* <<create-api-AP, Create an API using Anypoint Platform>>
+
If you want to create REST or SOAP APIs with OAS or WSDL definition, you use Anypoint Platform. You can manage your contracts and complex API configurations easily in Anypoint Platform.
* <<create-api-ASM-CLI, Create an API using Anypoint Service Mesh CLI>>
+

> If you want to create a simple HTTP API with no contracts or definitions, use the auto-creation feature of Anypoint Service Mesh. When you choose this option, Anypoint Service Mesh creates the API for you in Anypoint Platform using its powerful CLI tool. You do not have to manually log in to Anypoint Platform or manage your API. 

Reviewer 2: IMHO, It should talk about contracts here. You can use contracts even if you created your API with SM.

Author: Added the comment in the doc to address. Thanks!

+
Note that if you create REST or SOAP APIs using the auto-create feature of Anypoint Service Mesh, the APIs are still managed internally as an HTTP asset without any definition. 
+

> *Chirag,I have serious concerns for using the "auto-creation feature" terminology because nowhere in the CLI commands or the interface does it show it. If we just refer to it for marketing purpose but don't call it as such in the product, we must also omit it in the docs.* 

Reviewer 2: @chiragnarangmulesoft

Reviewer: auto-creation is a concept not just marketing fluff. It doesn't need to be called out in CLI or UI.

. In the *Application type* field, select the *Non-Mule application* radio button.
. Optionally, specify the *Implementation URI*, and click *Save*.
+
Your API is now created.

[[create-api-ASM-CLI]]
== Creating an API Using Anypoint Service Mesh CLI

Before you create the API, you must complete the task prerequisite.

*Task Prerequisite*

> Ensure that you obtain the environment ID and organization ID. You can perform this task from either API Manager or from Access Management. You require the <INSERT_PERMISSION> permission or <INSERT_ROLE> role to see to obtain these credentials.

Reviewer 2: We need to add the permissions part of who can create the API. I added a paragraph before on the Doc Changes after GA. Basically, to create the API you need `In order to create an API you will need to have at least Exchange Contributor for the Business group where you are creating the asset and Manage APIs configuration for the environment where you are creating the API Instance. `

Reviewer 2: But keep in mind that that is different from the permission `Ensure that you obtain the environment ID and organization ID.`

Reviewer 2: For the latter permission is just access to APIM.

Author: Done! Thanks!

| `assetId` | The ID of the assets that are a part of an organization. Supports alphanumeric characters without spaces. | Required | N/A
| `version` | The version of the asset on Exchange. *Needs to be semver* what is this chirag? | Required | N/A
| `instanceLabel` | The ID of the API instances of an Asset within a given environment. Supports alphanumeric characters without spaces. | Required | N/A
| `specTags` | Tags to be added to the asset on Exchange. |  Optional | N/A
| `instanceTags` | Tags to be added to the API instance on API Manager. |  Optional | N/A
| `user` | The username credential of the Anypoint Platform user, who is creating the API. | Required | N/A
| `password` | The password of the Anypoint Platform user, who is creating the API. | Required | N/A
|===
+
. Verify the <<provision-adapter-configure-service-mesh#adapter_status, adapter status>>:
+
`./service-mesh adapter list`
+
The adapter displays a Ready status.
. Optionally, you can verify that an asset is created on Exchange and an API is created on API Manager.

[[create_api-CRD]]
== Creating an API Using the CRD File

> To create an API in Anypoint Platform using the CRD file:

Reviewer 2: ```suggestion
To create an API in Anypoint Platform using a CRD:
```

Author: Done! Thanks!

. Create your adapter definition file and save it with a '.yaml' file extension:
+
[source,text,linenums]
----
apiVersion: servicecatalog.k8s.io/v1beta1
kind: ServiceInstance
metadata:
  name: <api name> # example: orders-api

>   namespace: <namespace name> # example: default

Reviewer 2: ```suggestion
  namespace: <namespace name> # example: default
  labels: 
    service-mesh.mulesoft.com/kind: api
```

Author: Done! Thanks!
